.github/workflows/release.yaml

```diff
@@ -6,69 +6,118 @@ on:
+      uses: actions/checkout@v2
       with:
-        ref: develop
+        ref: master
         token: ${{ secrets.ADMIN_GITHUB_TOKEN }}
+        fetch-depth: 0
     
     - name: Set up Github credentials
       run: |
         git config user.name Gaffer
         git config user.email github-actions@github.com
     
-    - name: Set version
+    - name: Set release version
       run: echo "RELEASE_VERSION=$(echo ${{ github.event.milestone.title }} | cut -c 2-)" >> $GITHUB_ENV
-    
-    - name: Update version
-      run: ./cd/update_app_version.sh ${RELEASE_VERSION}
+
+    - name: Set branch name
+      run: echo "BRANCH_NAME=$(git branch -a | grep $RELEASE_VERSION | tail -n 1 | cut -c 18-)" >> $GITHUB_ENV
+
+    - name: Set develop branch
+      if: ${{ !startsWith( env.BRANCH_NAME , 'alpha') }}
+      run: echo "DEVELOP_BRANCH=develop" >> $GITHUB_ENV
+
+    - name: Set v2-alpha branch
+      if: ${{ startsWith( env.BRANCH_NAME , 'alpha') }}
+      run: echo "DEVELOP_BRANCH=v2-alpha" >> $GITHUB_ENV
+
+    - name: Output branch name
+      id: branch
+      run: echo "::set-output name=branch_name::$BRANCH_NAME"
+
+    - id: develop_branch
+      run: echo "::set-output name=develop_name::$DEVELOP_BRANCH"
+
+    - name: Fail if no branch found
+      if: ${{ !env.BRANCH_NAME }}
+      run: exit 1
+
+    - name: Set alpha master branch
+      if: ${{ startsWith( env.BRANCH_NAME , 'alpha') }}
+      run: echo "MASTER_BRANCH=v2-alpha" >> $GITHUB_ENV
+
+    - name: Set master branch
+      if: ${{ !startsWith( env.BRANCH_NAME , 'alpha') }}
+      run: echo "MASTER_BRANCH=master" >> $GITHUB_ENV
+
+    - name: Checkout master
+      uses: actions/checkout@v2
+      with:
+        ref: ${{ env.MASTER_BRANCH }}
+        token: ${{ secrets.ADMIN_GITHUB_TOKEN }}
+        fetch-depth: 0
+
+    - name: Merge release into master
+      run: |
+        git checkout ${{ env.BRANCH_NAME }}
+        git checkout ${{ env.MASTER_BRANCH }}
+        git merge ${{ env.BRANCH_NAME }}
 
     - name: Push changes
       run: |
-        git commit -a -m "Release v${RELEASE_VERSION}"
         git tag v${RELEASE_VERSION}
-        git push origin --tags
+        git push origin v${RELEASE_VERSION}
         git push
 
   update-branches:
     runs-on: ubuntu-latest
     strategy:
       matrix:
         branch:
-        - master
+        - develop
         - gh-pages
     needs:
-    - create-git-tag
+    - create-release-tag
+
     steps:
     - uses: actions/checkout@v2
       with:
         ref: ${{ matrix.branch }}
         token: ${{ secrets.ADMIN_GITHUB_TOKEN }}
+        fetch-depth: 0
     
     - name: Set up Github credentials
       run: |
         git config user.name Gaffer
         git config user.email github-actions@github.com
   
-    - name: Update branch
+    # Skip develop if alpha release
+    - name: Merge release into branch
+      if: ${{ !((startsWith( needs.create-release-tag.outputs.branch_name , 'alpha')) && (matrix.branch == 'develop')) }}
```

Comment:
Can be moved to the job level instead.

Reply:
This one is slightly different as it is in a matrix and will run on the develop and gh-pages branches and I only wanted to skip merging develop but not gh-pages.

Reply:
You're right, this is different and I didn't look at it in proper detail. Couldn't the alpha part still be moved to the job level though? I'm not sure about the matrix, this step will only run if alpha is false and the branch is develop, if the branch is gh-pages then it doesn't look to me like it will run at all.

Reply:
Ah yes, brackets... It makes sense to me now. I agree this is better than 2 jobs.

Reply:
Lesser of 2 evils really